system-parachains/coretime/coretime-kusama/src/coretime.rs

```diff
@@ -0,0 +1,249 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use codec::{Decode, Encode};
+use cumulus_pallet_parachain_system::RelaychainDataProvider;
+use cumulus_primitives_core::relay_chain;
+use frame_support::{
+	parameter_types,
+	traits::{
+		fungible::{Balanced, Credit},
+		OnUnbalanced,
+	},
+};
+use pallet_broker::{
+	AdaptPrice, CoreAssignment, CoreIndex, CoretimeInterface, PartsOf57600, RCBlockNumberOf,
+};
+use parachains_common::{AccountId, Balance, BlockNumber};
+use sp_runtime::{
+	traits::{One, Saturating},
+	FixedU64,
+};
+use xcm::latest::prelude::*;
+
+pub struct CreditToCollatorPot;
+impl OnUnbalanced<Credit<AccountId, Balances>> for CreditToCollatorPot {
+	fn on_nonzero_unbalanced(credit: Credit<AccountId, Balances>) {
+		let staking_pot = CollatorSelection::account_id();
+		let _ = <Balances as Balanced<_>>::resolve(&staking_pot, credit);
+	}
+}
+
+/// A type containing the encoding of the coretime pallet in the Relay chain runtime. Used to
+/// construct any remote calls. The codec index must correspond to the index of `Coretime` in the
+/// `construct_runtime` of the Relay chain.
+#[derive(Encode, Decode)]
+enum RelayRuntimePallets {
```

Comment:
I am a little confused why we do this while we are in the same repository, and we have access to the specific `RuntimeCall` of the relay chain as well? There should be a way to fetch the index of a specific variant based on the name, without needing to re-hardcode number 74 here. 

Although truthfully a pallet index will really hardly change.

Reply:
I mean, I am not saying kusama-coretime runtime should have a kusama runtime nested into it, but there should be a way to bring over some of the constants and types of the kusama relay into individual parachains?